libraries/standard.js

```diff
@@ -1,8 +1,33 @@
 const lib = new Map();
 
+// check if read values are null, undefined or blank (empty string)
+// dst: destination object, data: data item being processed, 
+// data_field: field being processed, skip_errors: boolean, errors: array of bad data items
+const null_check_function = (dst, data, data_field, skip_errors, errors) => {
+  // data or data_field are null or undefined, or data is empty object
+  if (!data || Object.keys(data).length === 0 || !data_field) {
```

Comment:
the !data_field should be removed from here as its already being checked on line 18 and will cause a preemptive return here. The other comment about data_field should clear this up but DM me if you have questions

Reply:
Ok, I will make those changes! I had a question though, if the data is never being checked, except for if it's null at the start, does it need to be passed in as a parameter to this function?

Reply:
Aha good question. I missed this but you can change to pushing the entire data object to the errors array instead of just the field. Then we can report any bad data super accurately because the whole object is in the array, not just the field